### PR TITLE
arc-394:  Adds Token Info Fetching 

### PR DESCRIPTION
**Pull request by nicholaswma**
- Balances on dashboard are directly queried rather than querying for token info first
- Token Info are queried once a week with manual refreshing able to be called in the Token Settings page
- Balance will call token info if users are on Send page for accuracy 
- signDataItem auth screen will only query for Token Information if it's a `Transfer` (tighter params can be added)

### CODE REVIEW

File: src/components/dashboard/subsettings/TokenSettings.tsx [hunk 1]
```diff
@@ -73,35 +77,78 @@ export default function TokenSettings({ id }: Props) {
     });
   }
 
+  const refreshToken = async () => {
+    const defaultToken = defaultAoTokens.find((t) => t.processId === token.id);
```
Comment (pawanpaudel93) [inline src/components/dashboard/subsettings/TokenSettings.tsx:86]: Why not refresh token info for all ao tokens present ?

File: src/components/dashboard/subsettings/TokenSettings.tsx [hunk 2]
```diff
@@ -120,25 +167,59 @@ export default function TokenSettings({ id }: Props) {
...
           </AnimatePresence>
         </div>
       )}
-      <ButtonV2 fullWidth onClick={() => removeToken(id)}>
-        <TrashIcon style={{ marginRight: "5px" }} />
-        {browser.i18n.getMessage("remove_token")}
-      </ButtonV2>
+      <ButtonWrapper>
+        {isAoToken && (
+          <ButtonV2
```
Comment (pawanpaudel93) [inline src/components/dashboard/subsettings/TokenSettings.tsx:187]: Adding loading spinner to the Refresh Token will help to know if refresh is in progress.

File: src/tokens/aoTokens/ao.ts [hunk 3]
```diff
@@ -153,31 +166,47 @@ export function useAoTokens(): [TokenInfoWithBalance[], boolean] {
...
-                      process: AO_NATIVE_TOKEN_BALANCE_MIRROR,
-                      tags: [{ name: "Action", value: "Balance" }]
-                    });
-                    const balance = res.Messages[0].Data;
-                    if (balance) {
-                      return new Quantity(
-                        BigInt(balance),
-                        BigInt(12)
-                      ).toString();
+              const balance = Number(
```
Comment (pawanpaudel93) [inline src/tokens/aoTokens/ao.ts]: balance should not be converted to Number as precision is lost with it. Please, keep it as string.

File: src/components/dashboard/subsettings/TokenSettings.tsx (hunk 1 above)
Comment (nicholaswma) [reply src/components/dashboard/subsettings/TokenSettings.tsx:86]: I'm not opposed to that - I just did it this way since it's on an individual token page

Comment (pawanpaudel93) [reply src/components/dashboard/subsettings/TokenSettings.tsx:86]: @nicholaswma Yes, its on an individual token page but its refreshing for only tokens that are the `defaultTokens`. For `!defaultToken` the refresh is not handled. I mean can't we remove the default token check to refresh info for every token individually ?

Comment (nicholaswma) [reply src/components/dashboard/subsettings/TokenSettings.tsx:86]: good catch - I meant it to be the other way around, my intention was for default tokens to not have to be refreshed (this is essentially to not querying the ao token process)

**Pull request merged**